### ISSUE TITLE
💥 Crash on `+[PNDDiskBufferHelper getObjectsFromData:]` - Out of memory

### ISSUE THREAD

**Issue by msrutek-paylocity** [pendo-io/pendo-mobile-sdk#192]
## **Platform + Version**
iPadOS 18.0.0,  iPad (8th generation) 

## **SDK Version**
3.4.0

## **Framework**
Native, SwiftUI

## **Describe the bug**
Crash of the app.

## **To Reproduce**
Yet unknown.

## **Expected behavior**
No crash.

## **Stack Trace**

```
Fatal Exception: NSMallocException
Out of memory. We suggest restarting the application. If you have an unsaved document, create a backup copy in Finder, then try to save. 
0  CoreFoundation                 0x83f20 __exceptionPreprocess
1  libobjc.A.dylib                0x172b8 objc_exception_throw
2  CoreFoundation                 0x22ce8 -[__NSArrayM insertObject:atIndex:]
3  Pendo                          0x36978 -[PNDMutableArray addObject:]
4  Pendo                          0x104bd4 -[PNDTriggersManager addTrackEvent:]
5  Pendo                          0x59d98 __44+[PNDTrackEventsUtils sendTrack:properties:]_block_invoke
6  libdispatch.dylib              0x213c _dispatch_call_block_and_release
7  libdispatch.dylib              0x3dd4 _dispatch_client_callout
8  libdispatch.dylib              0xb400 _dispatch_lane_serial_drain
9  libdispatch.dylib              0xbf30 _dispatch_lane_invoke
10 libdispatch.dylib              0x16cb4 _dispatch_root_queue_drain_deferred_wlh
11 libdispatch.dylib              0x16528 _dispatch_workloop_worker_thread
12 libsystem_pthread.dylib        0x4934 _pthread_wqthread
13 libsystem_pthread.dylib        0x10cc start_wqthread
```

**Comment by TehilaTaub** [comment]
Hi @msrutek-paylocity 
We will be happy to have the full crash log if you have.
We are currently working on HF with a memory overload fix.
We will update you when it's out.

**Comment by MikePendo** [comment]
@msrutek-paylocity 
We just released new version 3.4.3, we had fixed significant memory issue that might arise on bad network. That might trigger or be the cause for the memory issue you have, any chance you could fetch the latest release and see and track the strange crashes once again. 
My comment applies to almost all issues you have found that related to memory, so I will copy paste that messages to the the relevant threads

**Comment by TehilaTaub** [comment]
Hi @msrutek-paylocity,
Please upgrade to 3.4.3 and if you notice similar issues, please open a new ticket with the crash logs.
Closing this one now.
